### PR TITLE
Removed inline js from excerpts.html

### LINKED ISSUE #4474: EPIC: Remove inline JS across all python templates
NOTE: This will enable OpenLibrary's JavaScript to be async and non-render blocking. When transitioning to async JavaScript it's vitally important we review the DOMContentLoaded event inside ol.analytics to make sure it is still firing.

Various templates across openlibrary add inline JavaScript files making it really hard to make changes to HTML or JS that is built in webpack without the potential to break things

The epic can be considered done when there are no matches for the following query:

```
# using [silver-searcher](https://github.com/ggreer/the_silver_searcher)
ag "(\<script type=\"text\/ja|\<script\>)"  --file-search-regex .html openlibrary/(macros|templates) -l --ignore "openlibrary/templates/site/body.html" --ignore openlibrary/templates/site/footer.html

```

This will allow us to drop the `window.q` handling code and begin to refactor/rewrite our JavaScript to a more modern framework.

### The How

The goal is to move JS out of HTML template files into JS files referenced inside `openlibrary/plugins/openlibrary/js/index.js`

Typically inline scripts are used when we need to pass data from the server to the client. Historically we've dynamically generated JavaScript

There are several ways to use this
* Use the JS equivalent function
* If a string use a data attribute, and read the value from the DOM in JS.
* If an object use a script tag with type="text/json+graph" following the example in openlibrary/templates/publishers/view.html calling `json_encode`

## Recommended steps

- Understand how the code works and where the code works before starting. Note some templates may be used on different pages.
- Remove the inline script in the HTML template and move to a file inside openlibrary/plugins/openlibrary/js/<feature name>.js
- Load <featurename>.js inside index.js
- Run `make js`
- Verify the page still works as before without any JS in the error console.
- Make sure the JS is loaded. See note below.
- Replace any references to `window.q.push` with `$(fn)`
- Document the new code explaining to your best understanding what it does and make sure it passes eslint
- If possible, make a video showing before/after of the code working (something like [loom](https://www.loom.com/) makes it easy)

Note: To avoid large amounts of JS in index.js we will import JS as required based on checks on the HTML.
For example
```
if (document.getElementById('modal-link')) {
        import(/* webpackChunkName: "patron_metadata" */ './patron-metadata')
            .then((module) => module.initPatronMetadata());
    }
```
Note 2: Some JS is duplicated in other files. If that's the case, the goal should be to simplify the existing code to avoid the duplication.

### Acceptance criteria
Pick a file or folder in the list below

- [x] #9176
- [x] openlibrary/templates/account/loans.html (medium) #8378 
- [ ] openlibrary/templates/swagger/swaggerui.html (medium) #8381
- [x] openlibrary/templates/account/ia_thirdparty_logins.html (easy) #8379
- [x] openlibrary/templates/type/i18n/edit.html #8336
- [x] openlibrary/templates/type/i18n/view.html #8336
- [x] openlibrary/templates/lib/edit_head.html (very easy) #8383
- [x] openlibrary/templates/books/breadcrumb_select.html (easy) #8380
- [x] openlibrary/templates/account/readinglog_stats.html (medium) (use data-config on div.readinglog-charts element instead) #8382
- [x] openlibrary/macros/TypeChanger.html #8374
- [x] openlibrary/macros/LoanStatus.html @jimchamp #8275
- [x] openlibrary/templates/lists/lists.html #2468
- [x] openlibrary/templates/lists/widget.html (#6159)
- [x] openlibrary/templates/merge/authors.html
- [x] openlibrary/templates/covers/change.html
- [x] openlibrary/templates/covers/saved.html
- [x] openlibrary/templates/covers/manage.html @drakene
- [x] openlibrary/templates/covers/add.html
- [x] openlibrary/templates/admin/case.html @RayBB #8260
- [x] openlibrary/templates/subjects.html  @Yashs911 
- [x] openlibrary/templates/admin/people/edits.html @drakene 
- [x] openlibrary/templates/admin/people/view.html @drakene 
- [x] openlibrary/templates/books/check.html @nileshtrivedi @CliftonMcCallum
- [x] openlibrary/templates/books/author-autocomplete.html @RayBB
- [x] openlibrary/templates/books/edit/addfield.html @RayBB #8261 (previously @cdrini #2700)
- [x] openlibrary/templates/books/edit/edition.html @RayBB
- [x] openlibrary/templates/books/edit.html @lephemere 
- [x] openlibrary/templates/books/edit/web.html  @lephemere
- [x] openlibrary/templates/books/edit/excerpts.html
- [x] openlibrary/templates/history.html
- [x] openlibrary/templates/type/author/view.html @Yashs911  #5262
- [x] openlibrary/templates/type/user/edit.html  @RayBB 
- [x] openlibrary/templates/type/user/view.html @RayBB 
- [x] openlibrary/templates/type/list/view_body.html #2468
- [x] openlibrary/templates/work_search.html
- [x] openlibrary/templates/lib/history.html
- [x] openlibrary/templates/site/stats.html @Yashs911 
- [x] openlibrary/templates/publishers/view.html
- [x] openlibrary/templates/account/borrow.html #5644
- [ ] Write a script to run in CI on PRs to prevent us from adding anymore inline js!

### Stakeholders
@jdlrobson 

If you want to help please comment issue for the file you'd like to work on.

### PR DESCRIPTION
**Pull request by nynaalekhya**
Closes #4474

Removed inline js from excerpts.html  and moved to edit.js

### Stakeholders

@jdlrobson

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/index.js [hunk 1]
```diff
@@ -91,74 +91,80 @@ jQuery(function () {
...
     }
 
     if (document.getElementById('modal-link')) {
-        import(/* webpackChunkName: "patron_metadata" */ './patron-metadata')
+        import (/* webpackChunkName: "patron_metadata" */ './patron-metadata')
             .then((module) => module.initPatronMetadata());
     }
 
+    if (document.getElementById('excerpts')) {
+        import ('./edit.js')
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:151]: You'll need to define a webchunk here `/* webpackChunkName: "patron_metadata" */`

File: openlibrary/plugins/openlibrary/js/index.js [hunk 2]
```diff
@@ -91,74 +91,80 @@ jQuery(function () {
...
     }
     if ($('script[type="text/json+graph"]').length > 0) {
-        import(/* webpackChunkName: "graphs" */ './graphs')
+        import (/* webpackChunkName: "graphs" */ './graphs')
             .then((module) => module.init());
     }
 
     if (window.READINGLOG_STATS_CONFIG) {
-        import(/* webpackChunkName: "readinglog_stats" */ './readinglog_stats')
+        import (/* webpackChunkName: "readinglog_stats" */ './readinglog_stats')
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:135]: please make sure to keep unrelated changes like this out of the final diff.

File: openlibrary/plugins/openlibrary/js/index.js [hunk 3]
```diff
@@ -91,74 +91,80 @@ jQuery(function () {
...
             .then(module => module.init(window.READINGLOG_STATS_CONFIG));
     }
 
     const pageEl = $('#page-barcodescanner');
     if (pageEl.length) {
-        import(/* webpackChunkName: "page-barcodescanner" */ './page-barcodescanner')
+        import (/* webpackChunkName: "page-barcodescanner" */ './page-barcodescanner')
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:141]: please make sure to keep unrelated changes like this out of the final diff.

File: openlibrary/plugins/openlibrary/js/index.js [hunk 4]
```diff
@@ -91,74 +91,80 @@ jQuery(function () {
...
             .then((module) => module.init());
     }
 
     if (document.getElementById('modal-link')) {
-        import(/* webpackChunkName: "patron_metadata" */ './patron-metadata')
+        import (/* webpackChunkName: "patron_metadata" */ './patron-metadata')
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:146]: please make sure to keep unrelated changes like this out of the final diff.

File: openlibrary/plugins/openlibrary/js/index.js [hunk 5]
```diff
@@ -91,74 +91,80 @@ jQuery(function () {
...
     }
 
+    if (document.getElementById('excerpts')) {
+        import ('./edit.js')
+            .then((module) => module.initEdit());
+    }
+
     if ($('#cboxPrevious').length) {
-        $('#cboxPrevious').attr({'aria-label': 'Previous button', 'aria-hidden': 'true'});
+        $('#cboxPrevious').attr({ 'aria-label': 'Previous button', 'aria-hidden': 'true' });
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/index.js:156]: please make sure to keep unrelated changes like this out of the final diff.

Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/index.js:156]: also applies to the next few lines in this file

**Review by jdlrobson — CHANGES_REQUESTED** [review]
I'm struggling to review this one as there are lots of cases where you have changed lines, but not actually tried functionality.

Please aim to get the diff of your changes as small as possible for reviews as it makes the life of the reviewer a lot easier, by allowing them to focus on the parts that have changed. I'll take another look at this when that's done.

**Comment by jdlrobson** [comment]
Hello @nynaalekhya do you have any questions? This patch has the right idea, and I'd love to see it merged but the unrelated changes make it hard to review.

**Comment by nynaalekhya** [comment]
Thank you for asking! I will make the suggested changes.

**Comment by nynaalekhya** [comment]
@jdlrobson, I have made the specified changes

**Review by jdlrobson — APPROVED** [review]
LGTM!

Tested on 
http://localhost:8080/books/OL24152177M/Robinson_Crusoe/edit

Screen Shot 2021-03-03 at 6 14 09 PM

Also confirmed feature code only loads on editor - not homepage. Great job with the feature detected import!

**Pull request merged**